test/libsolidity/SolidityOptimizer.cpp

```diff
@@ -665,6 +665,42 @@ BOOST_AUTO_TEST_CASE(optimise_constant_to_codecopy)
 	BOOST_CHECK_EQUAL(numInstructions(m_optimizedBytecode, Instruction::CODECOPY), 4);
 }
 
+BOOST_AUTO_TEST_CASE(byte_access)
+{
+	char const* sourceCode = R"(
+		contract C
+		{
+			function f(bytes32 x) public returns (byte r)
+			{
+				assembly { r := and(byte(x, 31), 0xff) }
+			}
+		}
+	)";
+	compileBothVersions(sourceCode);
```

Comment:
Do you think it makes sense to think about support for such comparisons in `soltest`? We could introduce an option (configuration of / options for tests need to be designed, though) and do the comparison plus the additional checks in the framework.

Reply:
The framework tests against the actual value and is run on both optimized and non-optimized code, so it already does that. We cannot compare the result of other calls because we cannot easily determine which other functions and valid input values exist.